src/PoCs/threeStepPopover.tsx

```diff
@@ -0,0 +1,105 @@
+import { Popover, ProgressStep, ProgressStepper, Button } from "@patternfly/react-core";
+import React, {
+  FunctionComponent,
+  useState,
+  VoidFunctionComponent,
+} from "react";
+
+type KafkaStatusPopoverProps = {
+  shouldStartOpen?: boolean;
+} & KafkaStatusPopoverBodyProps;
+
+/**
+*<strong> Introduction/Background </strong>
+*
+*This is a popover that can be reached from the Kafka Instance table while the status of the instance in question is “Creating”. A Kafka instance usually takes a few minutes to create. The purpose of this popover is to give the user some insights into the creation process of their instance while they wait for it to be available to use. 
+*
+*  -The text at the top explains that the instance will be ready shortly & suggests that the user creates a service account while they wait. (Provides a link to the “Connections tab" where this is done).
+*
+*  -The steps shown in the progress stepper are statuses that show in the api. They have been grouped together under the one overall “Creating status” in the table.There is some descriptive text accompanying each step too. When these steps are complete the instance will be ready for use
+*
+*  -The popover is only available while instance creation is in progress.
+*
+*  -After creation is complete the table displays the ready status and the popover is no longer available.
+*<br><br><br><br>
+*<h3>Pending</h3>
+*
+*A user has just selected to create a Kafka Instance. The creation of their instance is pending.
+*/
+export const KafkaStatusPopover: FunctionComponent<KafkaStatusPopoverProps> = ({
+  shouldStartOpen = false,
+  currentState,
+  children,
+}) => {
+  const [isVisible, setIsVisible] = useState(shouldStartOpen);
+
+  return (
+    <Popover
+      headerContent="Creating instance"
+      bodyContent={<KafkaStatusPopoverBody currentState={currentState} />}
+      isVisible={isVisible}
+      shouldOpen={() => setIsVisible(true)}
+      shouldClose={() => setIsVisible(false)}
+      position={'right'}
+      enableFlip={false}
+    >
+      {children}
+    </Popover>
+  );
+};
+
+type States = "pending" | "provisioning" | "preparing" | "ready" ;
+
+type KafkaStatusPopoverBodyProps = {
+  currentState: States;
+};
+export const KafkaStatusPopoverBody: VoidFunctionComponent<
+  KafkaStatusPopoverBodyProps
+> = ({ currentState }) => {
+  const order: States[] = ["pending", "provisioning", "preparing", "ready"];
+  const currentStep = order.findIndex((v) => v === currentState);
+  console.log(currentStep);
+  return (
+    <div>
+        <p>This will be ready shortly. In the meantime, you can create a service account under the <a href="http://www.google.com/">Connections tab.</a> </p>
+        <br></br>
```

Comment:
Using line breaks to control spacing isn't ideal. I'm sure there's a good explanation on why somewhere out there—my best explanation is that there are a set of PF space sizes that are intended to create consistency with regard to space between UI elements. 

Ideally, we would have space applied to the `<p>` elements, something like `margin-bottom` set to `var(--pf-global--spacer--md)`.  And applying this space to the `<p> above and also below with the current step info would provide the spacing we need above the progress stepper.

Usually, you would put text contents into a [`<TextContent>` component](https://www.patternfly.org/v4/components/text), but that component is problematic inside the popover. In all of our popovers, we want the font-size to be consistent, but if we place a `<TextContent>` component inside the popover, then the font-size of the TextContent component will override the font-size of the popover making the text larger than expected. 

Some other options: 

- Use a [PF utility class](https://www.patternfly.org/v4/utilities/spacing) `pf-u-mb-md` on both `<p>` elements to apply `margin-bottom` using the `md` space size
- Create a new css class for this component that provides `margin-bottom: var(--pf-global--spacer--md);` and add the class to both `<p>` elements. There is probably a class name format that we would want to use for this, but I'm not sure what our standards for custom class names are.

@riccardo-forina, @srambach  - Do you have any recommendations for an approach to take here?

Reply:
I'd say unless this is truly a one-off, I'd recommend creating a custom style. Since the utility classes use `!important`, they are a little heavy-handed so if you can avoid them, it's not a bad idea.
I was wondering about using the `<TextContent>` with a class and setting a base font-size scoped to that, but I don't see an easy way of doing this.